### ISSUE TITLE
Metric jsx-a11y/label-has-for fails for implicit labels

### ISSUE THREAD

**Issue by AlmeroSteyn** [evcohen/eslint-plugin-jsx-a11y#187]
Currently the plugin checks for `labels` without `htmlFor`. However implicit labeling also lead to accessible labels BUT will cause this metric to report.

Therefore the following validates correctly:
```
<label htmlFor="someId">Some label</label>
<input id="someId" />
```

But the following leads to an ESLINT error although it is accessible HTML:
```
<label>Some label
     <input id="someId" />
</label>
```

Could the metric be adjusted to allow for the second pattern?

**Comment by evcohen** [comment]
this has been spec-ed out but im not sure if it has been worked on yet: https://github.com/evcohen/eslint-plugin-jsx-a11y/issues/51#issuecomment-240620600

let me know if you think this solution works for you!

**Comment by AlmeroSteyn** [comment]
Oh dear, I completely missed the related issue! 

If the rule is configurable to allow for nesting it would indeed work. 

Thank you for responding.
